### PR TITLE
Migration to Swift 4

### PR DESCRIPTION
**Pull request by ghost**
@rolson @mhdostal please review

### CODE REVIEW

File: Toolkit/ArcGISToolkit/MeasureToolbar.swift [hunk 1]
```diff
@@ -93,11 +93,11 @@ class MeasureResultView : UIView{
         stackView.topAnchor.constraint(equalTo: topAnchor).isActive = true
         stackView.heightAnchor.constraint(equalToConstant: 32).isActive = true
         
-        valueLabel.setContentCompressionResistancePriority(500, for: .horizontal)
-        valueLabel.setContentHuggingPriority(1000, for: .horizontal)
+        valueLabel.setContentCompressionResistancePriority(UILayoutPriority(rawValue: 500), for: .horizontal)
+        valueLabel.setContentHuggingPriority(UILayoutPriority(rawValue: 1000), for: .horizontal)
```
Comment (rolson) [inline Toolkit/ArcGISToolkit/MeasureToolbar.swift]: i would change this one to `.required`

File: Toolkit/ArcGISToolkit/MeasureToolbar.swift [hunk 2]
```diff
@@ -93,11 +93,11 @@ class MeasureResultView : UIView{
...
         
-        unitButton.setContentCompressionResistancePriority(499, for: .horizontal)
-        unitButton.setContentHuggingPriority(1000, for: .horizontal)
+        unitButton.setContentCompressionResistancePriority(UILayoutPriority(rawValue: 499), for: .horizontal)
+        unitButton.setContentHuggingPriority(UILayoutPriority(rawValue: 1000), for: .horizontal)
```
Comment (rolson) [inline Toolkit/ArcGISToolkit/MeasureToolbar.swift]: i would change this one to `.required`

File: Toolkit/ArcGISToolkit/MeasureToolbar.swift [hunk 3]
```diff
@@ -376,18 +376,18 @@ public class MeasureToolbar: UIView, AGSGeoViewTouchDelegate {
...
         
         let c1 = resultView.leadingAnchor.constraint(greaterThanOrEqualTo: leftHiddenPlaceholderView.trailingAnchor, constant: space)
-        c1.priority = UILayoutPriorityRequired
+        c1.priority = UILayoutPriority.required
         
         // have to give this just below required, otherwise before the left and right views are setup in 
         // their proper locations we can get constraint errors
         let c2 = resultView.trailingAnchor.constraint(lessThanOrEqualTo: rightHiddenPlaceholderView.leadingAnchor, constant: -space)
-        c2.priority = UILayoutPriorityRequired-1
+        c2.priority = UILayoutPriority(rawValue: UILayoutPriority.RawValue(Int(UILayoutPriority.required.rawValue)-1))
```
Comment (rolson) [inline Toolkit/ArcGISToolkit/MeasureToolbar.swift]: i would just change this to 
`UILayoutPriority(rawValue: 999)`

**Review by rolson — CHANGES_REQUESTED** [review]

File: Toolkit/ArcGISToolkit/MeasureToolbar.swift [hunk 4]
```diff
@@ -93,11 +93,11 @@ class MeasureResultView : UIView{
         stackView.topAnchor.constraint(equalTo: topAnchor).isActive = true
         stackView.heightAnchor.constraint(equalToConstant: 32).isActive = true
         
-        valueLabel.setContentCompressionResistancePriority(UILayoutPriority(rawValue: 500), for: .horizontal)
-        valueLabel.setContentHuggingPriority(UILayoutPriority(rawValue: 1000), for: .horizontal)
+        valueLabel.setContentCompressionResistancePriority(.required, for: .horizontal)
```
Comment (rolson) [inline Toolkit/ArcGISToolkit/MeasureToolbar.swift]: this is wrong - 1000 is required

File: Toolkit/ArcGISToolkit/MeasureToolbar.swift [hunk 5]
```diff
@@ -93,11 +93,11 @@ class MeasureResultView : UIView{
...
+        valueLabel.setContentHuggingPriority(.required, for: .horizontal)
         
-        unitButton.setContentCompressionResistancePriority(UILayoutPriority(rawValue: 499), for: .horizontal)
-        unitButton.setContentHuggingPriority(UILayoutPriority(rawValue: 1000), for: .horizontal)
+        unitButton.setContentCompressionResistancePriority(.required, for: .horizontal)
```
Comment (rolson) [inline Toolkit/ArcGISToolkit/MeasureToolbar.swift]: this is wrong - 1000 is required

**Review by rolson — CHANGES_REQUESTED** [review]

**Comment by gdhakal** [comment]
@rolson made those changes. please take a look.

**Review by rolson — APPROVED** [review]

**Comment by rolson** [comment]
looks fine to me - please just make sure to test everything out

**Review by mhdostal — APPROVED** [review]

**Pull request merged**